### ISSUE TITLE
Enhance plot labels appearance (enhancement)

### ISSUE THREAD

**Issue by ceraolo** [OpenModelica/OpenModelica#10447]
### Description
I understand that the OM community of users has not the same sensitivity I have on graphical quality of plots. However, I hope the recommendations I set here at least don't have people against; they may be funded (along with other tickets) through a DFD from my university in a near future. 

The recomendations below have a special importance in case small plots are required, which occurs often, when we want to see several plots simultanteously in subwindows.

Consider the file Peaks.mo enclosed in ticket #10228, and the first picture in that ticket, reproduced below for convenience, along with its Dymola counterpart:
![image](https://user-images.githubusercontent.com/7373030/227761886-e94231a8-5a3b-4f9e-a5c3-1eaa08d29418.png)
![image](https://user-images.githubusercontent.com/7373030/227762517-c33703b4-8223-4871-8bd5-1bc9fd56fd1a.png)

I have some objections on this picture:
1. if we want the exponential form to be kept on all numbers (which I do not recommend), at least the following improvements should be done:
   . the numbers must be compacted. In this case, instead of e+04, much better would be e4, so that important horizontal space is saved.
  . all the numbers must be written with the same precision. So, instead of having e4 and 3.5e4 one should have 4.0e4 and 3.5e4, as it occurs with Dymola (Dymola spoils too much horizontal space for the vertical axis label, but this is Dymola's issue)
2. much better, however, would be for me to use the "e4" label ony once, as it occurs with MC's PlotXY (that I wrote), as in the following picture. I can give up this, if there is not enough agreement on my preference.

![image](https://user-images.githubusercontent.com/7373030/227762473-78786158-26db-47eb-a6d9-4274c86571b7.png)

When plots are made very small, additional issues occur. 
Compare the following two pictures made with OM and PlotXY respectively:
![image](https://user-images.githubusercontent.com/7373030/227762622-8409906a-99fd-4399-9ae2-109efb96f270.png)

![image](https://user-images.githubusercontent.com/7373030/227762631-993f1bde-aa24-4a2c-b20f-d16c000aa00d.png)

I see the following issues in OM's version:
1. the label size is too large in compaison with the general plot appearance
2. the number of vertical and horizontal labels is the same as for larger plots, with is suboptimal. With tiny plots the number of labels shold be reduced (I'm not asking for the label font size proportional to the whole plot. Just to reduce them slightly for very small plots, with some saturation so that the font size never falls below a minimum)
3. in this case in the horizontal axis the labels even overlap, which is very unconvenient.

So, I conclude that in general OM the plot quality, especially for tiny plots (which, as I said, occurs often when we want to see several plots simultanteously in subwindows), should be improved significantly. 

### Version and OS

 - OpenModelica Version: [e.g. `omc --version` or `Help->About OMEdit` from OMEdit]
 - OS: [e.g. Windows 10, 64 bit]
 - Versions of used Modelica libraries if applicable

**Comment by casella** [comment]
@ceraolo I fully agree that we must make the plot quality robust w.r.t. their size.

We can definitely look into this once we have completed the transition to the new instance-based interface, that has top priority now.

**Comment by casella** [comment]
Maybe we should avoid the unit ticks in the smaller figure.

**Comment by ceraolo** [comment]
Can ticks between axis labels be removed when distance between ticks is below  _x_ times the tick width?   (I suppose this width is usually 1 pixel, but probably on 4k screens is two pixels or more)

Consider, for instance the two plots in 
https://github.com/OpenModelica/OpenModelica/issues/10447#issuecomment-1816626157

I think in both of them the horizontal axis is ok,  the vertical axiis ticks on the small plot are useless (better was if they were not there), the vertical ticks on the larger plot are somehow borderline.
I estimate the _x_ factor above should be around 7-8. A few tests may help in finding the optimal value.

**Comment by ceraolo** [comment]
There are also worse examples.
Look at this:

![image](https://github.com/OpenModelica/OpenModelica/assets/7373030/5e4a93e6-2bf3-4082-9020-6080b2665e9d)

For me this looks terrible.
Here mention two issues (in addition to the unuseful ticks between vertical labels as per comment https://github.com/OpenModelica/OpenModelica/issues/10447#issuecomment-1816628989)
- the two row legend occupies too much vertical space (I reported this a few yyears ago, but at that time I was told that this was impossible to fix. Maybe now this is possible?)
- I have too many labels on the vertical axis for the allowble space

This is due to the two-row legend. I made an svg and the legend was single row, and got a much better result:
![a](https://github.com/OpenModelica/OpenModelica/assets/7373030/2fd33594-a4ef-45a5-b184-83c9b2a2cce2)

On this I would just object the unuseful ticks between vertical lables, and the strange fact that time written is in a different style than the numerical labels (courier?); this is specific to svg: not present in the screen plot.


Plots of this comment were made with v1.23.0-dev-168

**Comment by ceraolo** [comment]
In Nov 2023 the large part of this ticket's requiremement was solved (see https://github.com/OpenModelica/OpenModelica/issues/10447#issuecomment-1816626157). Just the issues of the number of numerical labels and of the height of two-row legend remained, which have both been recently solved (#15788).
Now we can close this ticket as well.
